packages/kbn-test/src/es/test_es_cluster.ts

```diff
@@ -137,5 +220,47 @@ export function createTestEsCluster(options: TestClusterFactoryOptions) {
 
       return format(parts);
     }
+
+    /**
+     * Returns a list of host URLs for the cluster. Intended for use
+     * when configuring Kibana's `elasticsearch.hosts` in a test.
+     *
+     * If the cluster has multiple nodes, each node URL will be included
+     * in this list.
+     */
```

Comment:
maybe we should throw if `getUrl` is called when `nodes` is specified? we can even implement different interfaces for cases with and without `nodes`

Reply:
> maybe we should throw if getUrl is called when nodes is specified

That's a good idea, will add.

> we can even implement different interfaces for cases with and without nodes

I still see value in `getHostUrls` as it supports both single and multi-node setups -- it just always returns an array. But yeah, we can update interfaces here to also omit `getUrl` when we have a multi-node config.

Reply:
> omit getUrl when we have a multi-node config.

Up to you. It's rather a "nice to have"

Reply:
> maybe we should throw if getUrl is called when nodes is specified

Let's add in the current PR

Reply:
Added both in the PR (throw when getUrl is called with multiple nodes specified, and also some TS enforcement which won't let you use getUrl if the node option is also provided).